[Chong Tze Zhao] iP

chongtzezhao opened the pull request:
# Atlas: Your AI Task Management Assistant 🤖

[Atlas](https://en.wikipedia.org/wiki/Atlas_(mythology)#:~:text=In%20Greek%20mythology%2C%20Atlas%20(%2F,in%20Roman%20mythology)%20and%20Perseus.) is a cutting-edge chatbot designed to revolutionize the way you manage your tasks and boost productivity. Here's what makes Atlas special:

- Natural language processing for intuitive task creation
- Smart prioritization algorithms
- Seamless integration with popular calendar apps
- Real-time collaboration features

## Key Features

1. **Voice-activated task creation**
2. **Intelligent task categorization**
3. **Customizable reminders and notifications**
4. **Detailed analytics and progress tracking**

## Development Roadmap

- [x] Implement core task management features
- [ ] Develop natural language processing module
- [x] Design user interface
- [ ] Integrate with third-party apps
- [ ] Conduct initial user testing

> "Atlas can share the weight of my problems" - Anon User

Atlas utilizes state-of-the-art `command line` algorithms to continually improve its performance and adapt to your unique task management style. Wait sorry, I meant that you have to adapt to Atlas.

Experience the future of productivity with Atlas - your tasks will never be the same!

**Bold** choices lead to *italic* success, and Atlas helps you ~~procrastinate~~ accelerate your journey to achievement.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chongtzezhao committed:
renamed, greet, exit

chongtzezhao committed:
made chatbot echo

chongtzezhao committed:
Added ability to store tasks in a list

chongtzezhao committed:
Added: ability to mark status of tasks

chongtzezhao committed:
Added: implemented functionality of ToDo, Deadline, Event

chongtzezhao committed:
Updated: Atlas introduction. Added: test files

chongtzezhao committed:
Added: Error handling

chongtzezhao committed:
Added: ability to delete tasks

chongtzezhao committed:
Added: testing for deletion of tasks

chongtzezhao committed:
Add ability to save and load tasks on local file storage

chongtzezhao committed:
Merge branch 'branch-Level-7'

chongtzezhao committed:
Update by,from,to to only accept dates and stores dates internally

chongtzezhao committed:
Merge branch 'branch-Level-8'

Johnwz123 commented on src/main/DateTime.java:
```diff
@@ -0,0 +1,11 @@
+import java.time.format.DateTimeFormatter;
+
+public class DateTime {
+    public static final DateTimeFormatter dmy = DateTimeFormatter.ofPattern("dd MMM yyyy");
```
Perhaps you could use a more descriptive name.
```suggestion
    public static final DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("dd MMM yyyy");
```

Johnwz123 commented on src/main/java/Atlas.java:
```diff
@@ -0,0 +1,335 @@
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+import java.util.Arrays;
+import java.time.LocalDate;
```
The ordering of import statements should be consistent.

Johnwz123 commented on src/main/java/Atlas.java:
```diff
@@ -0,0 +1,335 @@
+        botMessage(SEP);
+    }
+
+    private static void taskMark(int index, boolean status) throws ArrayIndexOutOfBoundsException {
```
```suggestion
    private static void markTask(int index, boolean isDone) throws ArrayIndexOutOfBoundsException {
```

Johnwz123 commented on src/main/java/Atlas.java:
```diff
@@ -0,0 +1,335 @@
+        tasks.get(index).setDone(status);
+    }
+
+    private static void taskMarkLog(int index, boolean status) {
```
```suggestion
    private static void markTaskLog(int index, boolean isDone) {
```

Johnwz123 commented on src/main/java/Atlas.java:
```diff
@@ -0,0 +1,335 @@
+
+public class Atlas {
+    private static final String BOT_INDENT = "\t";
+    private static final String SEP = "___________________________________________";
```
```suggestion
    private static final String SEPARATOR = "___________________________________________";
```

Johnwz123 reviewed (commented):
Few minor comments with regards to names.

chongtzezhao committed:
refactored code to separate functionalities

saajidshaik02 commented on .gitignore:
```diff
@@ -3,6 +3,12 @@
 /out/
 /*.iml
 
+# class files
```
Its good practice to use git.ignore for class and data files!

saajidshaik02 commented on src/main/java/Main.java:
Good OOP principles used!

saajidshaik02 commented on src/main/java/impl/Command.java:
perhaps enum should be capitalized?

saajidshaik02 commented on src/main/java/impl/TaskStorageImpl.java:
```diff
@@ -0,0 +1,246 @@
+    }
+
+    private TaskStorageResult<Command> addToDo(String[] inputParts) {
+        String description = arrayJoin(inputParts, 1, inputParts.length);
```
should document why the number 1 is used

saajidshaik02 commented on src/main/java/impl/TaskStorageImpl.java:
```diff
@@ -0,0 +1,246 @@
+        return newTask;
+    }
+
+    /* ----------------------------- HELPER ----------------------------- */
```
nice that u used comments to make ur code more readable!

saajidshaik02 reviewed (commented):
LGTM. Looks very well designed, jiayou!

chongtzezhao committed:
refactor structure into packages

chongtzezhao committed:
Merge branch 'add-gradle-support'

chongtzezhao committed:
Add Junit Testing

chongtzezhao committed:
Add javadoc to most public methods

chongtzezhao committed:
updated files to fit the coding standard

chongtzezhao committed:
add find capabilities

chongtzezhao committed:
Merge branch 'branch-A-JavaDoc'

chongtzezhao committed:
Merge branch 'branch-A-CodingStandard'

chongtzezhao committed:
Merge branch 'branch-Level-9'

chongtzezhao committed:
Add checkstyle

chongtzezhao committed:
add gui boilerplate

chongtzezhao committed:
Merge branch 'branch-A-CheckStyle'

chongtzezhao committed:
Merge branch 'branch-Level-10'

chongtzezhao committed:
Add JAR to "artifacts"

chongtzezhao committed:
Add GUI

chongtzezhao committed:
Merge branch 'branch-Level-10'

chongtzezhao committed:
generated jar file for gui

chongtzezhao committed:
Add assertions for improved robustness
The TaskStorageImpl class lacked internal checks to ensure method
preconditions and class invariants. This could lead to subtle bugs
and make debugging more difficult.
Let's add assertions throughout the TaskStorageImpl class to verify
important conditions and catch programming errors early. Specifically:

Add null checks for critical objects and parameters
Verify minimum array lengths for input parameters
Ensure the integrity of the tasks list and file path

These assertions will help catch issues during development and testing,
making the code more robust and easier to maintain. They are not meant
for production-level input validation, as assertions are typically
disabled in production environments.
Key changes include:

Constructor: Assert FILE_PATH is not null
getTasks(), saveTasks(): Assert tasks list is not null
addTask(): Assert inputParts and command are not null
Method-specific assertions for addToDo, addEvent, addDeadline
deleteTask(), findTasks(), setTaskDone(): Verify inputParts
loadTasks(): Ensure FILE_PATH is not null before loading
deserialize(): Check that input line is not null or empty

These changes align with best practices for defensive programming and
will help catch potential issues earlier in the development process.

chongtzezhao committed:
Improve code quality

chongtzezhao committed:
Merge pull request #2 from chongtzezhao/branch-A-Assertions

Branch A Assertions

chongtzezhao committed:
Merge pull request #3 from chongtzezhao/branch-A-CodeQuality

Branch A Code Quality

chongtzezhao committed:
Add sorting capability to when listing dated tasks

chongtzezhao committed:
Merge branch 'branch-C-Sort'

chongtzezhao committed:
Add UserGuide

chongtzezhao committed:
Add UI screenshot

chongtzezhao committed:
Add GUI jar

chongtzezhao committed:
Update User Guide

chongtzezhao committed:
Fix User Guide: UI.png

chongtzezhao committed:
Rename UI.png to Ui.png

chongtzezhao committed:
Fix TaskStorage filepath bug, update user guide

chongtzezhao committed:
Improved README

chongtzezhao committed:
Merge branch 'branch-improve-readme'

chongtzezhao committed:
Fix formatting for UG

chongtzezhao committed:
Merge branch 'branch-improve-readme'

chongtzezhao committed:
Style: Improve Javadocs

chongtzezhao committed:
Merge branch 'branch-improve-javadocs'

chongtzezhao committed:
Logic: Add check that "to" is after "from" date

chongtzezhao committed:
Merge branch 'branch-amend-behaviour'

chongtzezhao committed:
Style: Remove dead code

chongtzezhao committed:
Style: Fix indentation

chongtzezhao committed:
Merge branch 'branch-improve-code-quality'

chongtzezhao committed:
Merge branch 'master' into branch-improve-readme

chongtzezhao committed:
Docs: Add known issues

chongtzezhao committed:
Merge branch 'branch-improve-readme'

chongtzezhao committed:
Docs: Update formatting of known issues

chongtzezhao committed:
Merge branch 'branch-improve-readme'

chongtzezhao committed:
Docs: Fix grammar

chongtzezhao committed:
Merge branch 'branch-improve-readme'
